horiz_interp/include/horiz_interp.inc

```diff
@@ -517,25 +290,29 @@ contains
    method = 'bilinear'
    if(present(interp_method)) method = interp_method
 
+   if( .not. allocated(Interp % HI_KIND_TYPE_)) allocate (Interp % HI_KIND_TYPE_)
```

Comment:
Since I added this type to hold the reals in the interp derived type, it needs to get allocated in the horiz_interp_*_new routines if its not already (this way it only allocates for r4 or r8 depending on the macro). It gets deallocated in the horiz_interp_*_del routines.